### PR TITLE
Update libcurl to 7.78.0

### LINKED ISSUE #6449: [request] libcurl/7.78.0
### Package Details
  * Package Name/Version: **libcurl/7.78.0**
  * Changelog: **https://curl.se/changes.html#7_78_0**

The above mentioned version is newly released by the upstream project and not yet available as a recipe. Please add this version.

### PR DESCRIPTION
**Pull request by dheater**
Specify library name and version:  **libcurl/7.78.0**

This is an update to the the latest libcurl release.

`libmetalink` support has been removed so there is an update to ignore that flag for this and newer `libcurl` updates

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Closes  #6449

### CODE REVIEW

**Comment by SpaceIm** [comment]
Please consider https://github.com/dheater/conan-center-index/pull/5

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -313,6 +312,13 @@ def _get_configure_command_args(self):
         if self._has_zstd_option:
             params.append("--with-zstd={}".format(yes_no(self.options.with_zstd)))
 
+        # Support for metalink was removed in version 7.78.0 https://github.com/curl/curl/pull/7176
+        if tools.Version(self.version) >= "7.78.0":
+            if self.options.with_libmetalink:
+                raise ConanInvalidConfiguration("Support for metalink was removed in libcurl version 7.78.0")
```
Comment (prince-chrismc) [inline recipes/libcurl/all/conanfile.py:318]: typically we have this in configure and delete the option... no need to raise

```suggestion
        if tools.Version(self.version) >= "7.78.0":
            if self.options.with_libmetalink:
                raise ConanInvalidConfiguration("Support for metalink was removed in libcurl version 7.78.0")
```

Comment (dheater) [reply recipes/libcurl/all/conanfile.py:318]: @prince-chrismc Suggestion is the same as the original. Mistake?

**Comment by SpaceIm** [comment]
https://github.com/conan-io/conan-center-index/pull/6452#issuecomment-884457335

**Comment by SpaceIm** [comment]
Why don't you simply merge my PR?

**Comment by dheater** [comment]
> Why don't you simply merge my PR?

Sorry @prince-chrismc . I didn't see it until after my last commit.  I made a comment on it.

> I think we want to raise an invalid config error if a user still has with-libmetalink option specified and updates to >= 7.78.0 to let the user know

**Comment by SpaceIm** [comment]
> > Why don't you simply merge my PR?
> 
> Sorry @prince-chrismc . I didn't see it until after my last commit. I made a comment on it.
> 
> > I think we want to raise an invalid config error if a user still has with-libmetalink option specified and updates to >= 7.78.0 to let the user know

I'm not @prince-chrismc 😛 , and my PR suggests another approach. See dheater#5.

**Comment by dheater** [comment]
> > > Why don't you simply merge my PR?
> > 
> > 
> > Sorry @prince-chrismc . I didn't see it until after my last commit. I made a comment on it.
> > > I think we want to raise an invalid config error if a user still has with-libmetalink option specified and updates to >= 7.78.0 to let the user know
> 
> I'm not @prince-chrismc 😛 , and my PR suggests another approach. See [dheater#5](https://github.com/dheater/conan-center-index/pull/5).

Oops! @SpaceIm I forgot to hit "Submit" on my comments.

I think we want to raise an error to the user if they specify `with-metalink=True` for versions that don't support it.  Looks like in your PR it gets silently dropped.

**Comment by SpaceIm** [comment]
I understand what you mean, but it's not how recipe options are handled usually in CCI (for example when we create new subfolders to handle big changes in upstream build system, we don't try to keep the same options if they have been removed upstream).
libcurl has removed this option since libcurl 7.78.0, I believe we should do the same.

If someone tries to explicitly set a delete option, it will raises an error anyway I think.

**Comment by dheater** [comment]
> If someone tries to explicitly set a deleted option, it will raises an error anyway I think.

You are correct @SpaceIm . And I confirmed that `with_libmetalink=True` still builds with older version so I've merged. Thanks!

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**